docker-compose.yml

```
@@ -1050,6 +1051,30 @@ services:
       /bin/bash -c "
         /arrow/ci/scripts/r_sanitize.sh /arrow"
 
+  ubuntu-r-valgrind:
+    # Only 18.04 and amd64 supported
+    # Usage:
+    #   docker-compose build ubuntu-r-valgrind
+    #   docker-compose run ubuntu-r-valgrind
+    image: ${REPO}:amd64-ubuntu-18.04-r-valgrind-r
+    build:
+      context: .
+      dockerfile: ci/docker/linux-r.dockerfile
+      args:
+        base: wch1/r-debug:latest
+        r_bin: RDvalgrind
+    environment:
+      <<: *ccache
+      # AVX512 not supported by Valgrind (similar to ARROW-9851) some runners support AVX512 and some do not
+      # so some build might pass without this setting
+      ARROW_USER_SIMD_LEVEL: "AVX2"
+      ARROW_RUNTIME_SIMD_LEVEL: "AVX2"
```

Since you're using the R package's bundled build script, I think this is how you need to pass in extra args: 

```suggestion
      EXTRA_CMAKE_FLAGS: "-DARROW_USER_SIMD_LEVEL=AVX2 -DARROW_RUNTIME_SIMD_LEVEL=AVX2"
```

https://github.com/apache/arrow/blob/master/r/inst/build_arrow_static.sh#L75

---

`ARROW_USER_SIMD_LEVEL` isn't a CMake option, which is the whole point. You use the same binary and change the instruction set at runtime.

---

Thanks, suggestion amended